giskard/core/model_validation.py

```diff
@@ -14,11 +14,14 @@
 from giskard.models.base import BaseModel, WrapperModel
 from ..utils import fullname
 from ..utils.analytics_collector import analytics, get_dataset_properties, get_model_properties
+from .dataset_validation import validate_optional_target
 
 
 @configured_validate_arguments
 def validate_model(model: BaseModel, validate_ds: Optional[Dataset] = None, print_validation_message: bool = True):
     try:
+        if model.meta.model_type != SupportedModelTypes.TEXT_GENERATION and validate_ds is not None:
+            validate_optional_target(validate_ds)
```

Comment:
Why do we need this in `validate_model`???

Reply:
we should refactor `model_validation` and `dataset_validation` into one common `validation`, but the way the code is structured right now, I needed `model.meta.model_type` to do `validate_optional_target` for non-LLM models.

Reply:
Let's just fix the llm-warning issue for now and deal with the refactoring post-release, I added a card here: https://linear.app/giskard/issue/GSK-2064/refactor-model-and-dataset-validation